src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -817,8 +849,15 @@ protected RoutingRequest buildRequest() throws ParameterException {
         if (geoidElevation != null)
             request.geoidElevation = geoidElevation;
 
-        if (pathComparator != null)
+        setEarliestTransportationNetworkCompanyEta(request, router, companies);
```

Comment:
Calling external APIs when build a request is problematic/unexpected. I think this is the responsibility of the `RoutingService`. In this case the RoutingResource should build a request, and the delegate the execution. An example of why this is problematic is that the `RoutingService` may split the request into sub-tasks and run them in parallel, then all tasks have to wait on this one, even the one that do not depend on it.